test/hotspot/gtest/metaspace/test_freeblocks.cpp

```diff
@@ -91,6 +91,17 @@ class FreeBlocksTest {
     }
   }
 
+  void deallocate_all() {
+    while (_allocations != nullptr) {
+      allocation_t* a = _allocations;
+      _allocations = a->next;
+      check_marked_range(a->p, a->word_size);
+      _freeblocks.add_block(a->p, a->word_size);
+      delete a;
+      DEBUG_ONLY(_freeblocks.verify();)
+    }
+  }
```

Comment:
Can you reshape this a bit pls: let deallocate_top return bool (false if nothing removed) and do
```
void deallocate_all() {
  while (!deallocate_top());
}
```

Reply:
I'm a bit concerned about runtime. Does the runtime for these tests increase, since the verification on every deallocation is a bit overboard?

Reply:
It's not noticeable. Restructed the code.